src/commands/generate.ts

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as path from 'path';
+import * as flags from '../lib/flags';
+import * as frameGenerator from '../lib/controllers/frame/frame-generator';
+import * as frameTemplate from '../lib/controllers/frame-template';
+
+import { Command } from '@oclif/command';
+import { ConfigManifest } from '../lib/controllers/config-manifest/config-manifest';
+import { createConfigStore } from '../lib/controllers/config-store/config-store';
+import { ConfigurationManager } from '../lib/controllers/configuration-manager/configuration-manager';
+import {
+	loadEnvironment,
+	EnvironmentContext,
+} from '../lib/controllers/environment/environment';
+import {
+	filesystemExportAdapter,
+	InvalidOutputDirectory,
+} from '../lib/controllers/frame/adapter/filesystem';
+import { mustacheRenderer } from '../lib/controllers/frame-template/renderer/mustache';
+import { ConfigStoreError } from '../lib/error-types';
+
+/**
+ * Generate Command class
+ * The init command is used for generating deployment artifacts, using an environment configuration.
+ * If an environment configuration doesn't exist, or is out of sync, it's generated/synced.
+ */
+export default class Generate extends Command {
+	static description = 'Generate a Frame from an Environment';
+
+	static flags = {
+		environmentPath: flags.environmentPath,
+		outputPath: flags.outputPath,
+		target: flags.target,
+	};
+
+	async run() {
```

Comment:
This is an `async` function already so it should be easy enough to make the `*Sync` calls within it async instead

Reply:
Agreed. Should be done now.